engine/table/src/main/java/io/deephaven/engine/table/impl/WhereListener.java

```diff
@@ -214,25 +211,26 @@ void setFinalExecutionStep() {
     ListenerFilterExecution makeFilterExecution() {
         final ModifiedColumnSet sourceModColumns = getSourceModifiedColumnSet();
         final boolean runModifiedFilters = filterColumns == null || sourceModColumns.containsAny(filterColumns);
-        return new ListenerFilterExecution(recorder.getAdded(), 0, recorder.getAdded().size(),
-                recorder.getModified(), 0, recorder.getModified().size(), null,
-                runModifiedFilters, sourceModColumns, 0);
+        return new ListenerFilterExecution(recorder.getAdded(), recorder.getModified(),
+                runModifiedFilters, sourceModColumns);
     }
 
     class ListenerFilterExecution extends AbstractFilterExecution {
+        private final JobScheduler jobScheduler;
+
         private ListenerFilterExecution(
                 final RowSet addedInput,
-                final long addStart,
-                final long addEnd,
                 final RowSet modifyInput,
-                final long modifyStart,
-                final long modifyEnd,
-                final ListenerFilterExecution parent,
                 final boolean runModifiedFilters,
-                final ModifiedColumnSet sourceModColumns,
-                final int filterIndex) {
-            super(WhereListener.this.sourceTable, WhereListener.this.filters, addedInput, addStart, addEnd, modifyInput,
-                    modifyStart, modifyEnd, parent, false, runModifiedFilters, sourceModColumns, filterIndex);
+                final ModifiedColumnSet sourceModColumns) {
+            super(WhereListener.this.sourceTable, WhereListener.this.filters, addedInput, modifyInput,
+                    false, runModifiedFilters, sourceModColumns);
+            // Create the proper JobScheduler for the following parallel tasks
+            if (getUpdateGraph().parallelismFactor() > 1) {
```

Comment:
Need to check both `permitParallelization` and `QueryTable.FORCE_PARALLEL_WHERE`.